src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * Interactivity API: WP_Interactivity_API_Directives_Processor class.
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ * @since 6.5.0
+ */
+
+/**
+ * Class used to iterate over the tags of an HTML string and help process the
+ * directive attributes.
+ *
+ * @since 6.5.0
+ *
+ * @access private
+ */
+class WP_Interactivity_API_Directives_Processor extends WP_HTML_Tag_Processor {
```

Comment:
Prevent extending since not intended.

```suggestion
final class WP_Interactivity_API_Directives_Processor extends WP_HTML_Tag_Processor {
```

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5953/commits/1c735211a3d914e6ca3e488753c4583d0e5f4e82.